### PR TITLE
added ehcache config

### LINKED ISSUE #128: Add Spring Caching and some Cache implementation
We should prepare a caching implementation that can be used with Spring's Caching abstraction so that we can just use it whereever we please.

@outcobra/developers What implementation should we use?

### PR DESCRIPTION
**Pull request by needToRoll**
This closes #128 
Summary of changes:
- ​added caching dependencies to classpath
- added ehcache.xml
- Added CacheConfig (to create caches) 

I made sure to:
- [x] Add and / or update tests
- [x] Run the code formatter (`CTRL + ALT + L`)
- [x] Update any documentation (javadoc, domain model, erd, ...)
- [x] Follow the contribution guidelines
- [x] Update CI script and configs on Jenkins (`Jenkinsfile`, `auth0.yml`, `application.yml`, etc.)

Please review @jmesserli / cc: @outcobra/developers.

### CODE REVIEW

File: backend/src/main/kotlin/outcobra/server/util/RepositoryLocator.kt [hunk 1]
```diff
@@ -30,7 +30,7 @@ class RepositoryLocator @Inject constructor(val context: ApplicationContext) {
      * @see [BeanFactory.getBean]
      * @since 1.0.0
      */
-    @Cacheable(REPO_FOR_NAME)
+    @Cacheable(REPO_FOR_NAME, key = "entityName")
```
Comment (jmesserli) [inline backend/src/main/kotlin/outcobra/server/util/RepositoryLocator.kt]: This should be the default value, no need to specify it again imo

File: backend/src/main/kotlin/outcobra/server/util/RepositoryLocator.kt [hunk 2]
```diff
@@ -52,7 +52,7 @@ class RepositoryLocator @Inject constructor(val context: ApplicationContext) {
      * @throws NoRepositoryFoundException if the requested repository could not be found or has an illegal type
      * @since 1.0.0
      */
-    @Cacheable(REPO_FOR_ENTITY)
+    @Cacheable(REPO_FOR_ENTITY, key = "#entityClass.getSimpleName()")
```
Comment (jmesserli) [inline backend/src/main/kotlin/outcobra/server/util/RepositoryLocator.kt]: We should use `getCanonicalName` just to be safe, same for the second one

**Comment by needToRoll** [comment]
Tests are green, merge please

Btw what's up with this license error?

**Comment by jmesserli** [comment]
I don't like the fix, we'll look at it tomorrow

**Comment by needToRoll** [comment]
This is the recommended way to do it.
There is a sample in the docs doing this.
But let's talk about it face to face

**Comment by needToRoll** [comment]
@jmesserli is this merge ready?

**Pull request merged**